File: src/main/java/com/arcao/geocaching4locus/authentication/util/DeviceInfoFactory.java
package com.arcao.geocaching4locus.authentication.util;

import android.content.Context;
import android.os.Build;

import com.arcao.geocaching.api.data.DeviceInfo;
import com.arcao.geocaching4locus.App;

public final class DeviceInfoFactory {
	public static DeviceInfo create(Context context) {
		App app = App.get(context);

		return DeviceInfo.builder()
				.applicationCurrentMemoryUsage(0)
				.applicationPeakMemoryUsage(0)
				.applicationSoftwareVersion(app.getVersion())
				.deviceManufacturer(Build.MANUFACTURER)
				.deviceName(Build.MODEL)
				.deviceOperatingSystem(Build.VERSION.RELEASE)
				.deviceTotalMemoryInMb(0)
				.deviceUniqueId(app.getDeviceId())
				.build();
	}
}

File: app/src/main/java/com/arcao/geocaching4locus/authentication/util/DeviceInfoFactory.kt
package com.arcao.geocaching4locus.authentication.util

import android.os.Build
import com.arcao.geocaching.api.data.DeviceInfo
import com.arcao.geocaching4locus.App

class DeviceInfoFactory(private val app: App) {
    @JvmName("create")
    operator fun invoke(): DeviceInfo = DeviceInfo.builder()
        .applicationCurrentMemoryUsage(0)
        .applicationPeakMemoryUsage(0)
        .applicationSoftwareVersion(app.version)
        .deviceManufacturer(Build.MANUFACTURER)
        .deviceName(Build.MODEL)
        .deviceOperatingSystem(Build.VERSION.RELEASE)
        .deviceTotalMemoryInMb(0f)
        .deviceUniqueId(app.deviceId)
        .build()
}
